app/Http/Controllers/EventOrdersController.php

```diff
@@ -220,112 +215,131 @@ public function postCancelOrder(Request $request, $order_id)
             ]);
         }
 
+        /** @var Order $order */
         $order = Order::scope()->findOrFail($order_id);
+        /** @var Collection $attendees */
+        $attendees = Attendee::findFromSelection($request->get('attendees'));
+        $nonCancelledOrderAttendees = $order->getAllNonCancelledAttendees();
+
+        // Get the full order amount, tax and booking fees included
+        $organiserAmount = new Money($order->organiser_amount, $order->getEventCurrency());
+        Log::debug(sprintf("Total Order Value: %s", $organiserAmount->display()));
+        $refundedAmount = new Money($order->amount_refunded, $order->getEventCurrency());
+        Log::debug(sprintf("Already refunded amount: %s", $refundedAmount->display()));
+        $maximumRefundableAmount = $organiserAmount->subtract($refundedAmount);
+        Log::debug(sprintf("Maxmimum refundable amount: %s", $maximumRefundableAmount->display()));
+        /*
+         * NOTE: Calculates the refund amount by using the order total (tax incl) and dividing it by the amount of
+         * attendees the user has selected. This will evently spread the tax refund per attendee.
+         *
+         * If only one attendee lives in an order then its safe to assume a full refund can be made.
+         */
+
+        $refundAmount = $maximumRefundableAmount->multiply($attendees->count())
+            ->divide($nonCancelledOrderAttendees->count())->floor();
```

Comment:
Lets take the following scenario

Ticket 1 £400
Ticket 2 £400
Ticket 3 £150

Total order ex vat is £950. Total order with V.A.T is £1140.00.

I refund a single ticket of £400. 

The above calculation results in £380 however the ticket price is £480. The above calculation will work refunding the order as a whole but I think that’s quite confusing if the intention is to refund one ticket. 

I think this will need to change to refund the individual ticket as a whole

Reply:
Busy looking at this as well. Thanks for reporting 🙏